docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3795,34 +3803,56 @@ the `GetForwardingPipelineConfig RPC`. The request is defined as:
 
 ~ Begin Proto
 message GetForwardingPipelineConfigRequest {
+  enum ResponseType {
+    ALL = 0;
+    COOKIE_ONLY = 1;
+    P4INFO_ONLY = 2;
+  }
   uint64 device_id = 1;
+  ResponseType response_type = 2;
 }
 ~ End Proto
 
 The `device_id` uniquely identifies the target P4 device. An `INVALID_ARGUMENT`
 error is returned if the `device_id` is not recognized by the P4Runtime server.
 
-The response contains the `P4ForwardingPipelineConfig` for the specified device:
+The `response_type` is used to specify which fields to populate in the response,
+its value can be one of:
+
+* `ALL`: returns a `ForwardingPipelineConfig` with all fields
+  set as stored by the target. This is the default behaviour if the
+  `response_type` field is not set.
+
+* `COOKIE_ONLY`: reply by setting only the `cookie` field in the
+  `ForwardingPipelineConfig`, omitting all other fields. This mechanisms can be
+  used by a controller to verify that a config is the expected one, while
+  minimizing the amount of data in the response message.
+
+* `P4INFO_ONLY`: reply by setting only the `p4info` field.
```

Comment:
As I commented in the proto source, I think this mode should probably include the cookie (all modes should proabbly include the cookie IMO)

Reply:
Need to account for older versions, as I described above.

Reply:
Changed modes to always return the cookie (is present). Older versions should be supported by `ForwardingPipelineConfig.Cookie` sub-message.